Tests/Unit/Extend/Controller/OrderControllerTest.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+use OxidEsales\Eshop\Application\Model\User;
+use OxidEsales\Eshop\Core\DatabaseProvider;
+use OxidEsales\Eshop\Core\Field;
+use OxidEsales\Eshop\Core\Price;
+use OxidEsales\Eshop\Core\Registry;
+
+use Wirecard\Oxid\Extend\Controller\OrderController;
+use Wirecard\Oxid\Extend\Model\Basket;
+
+class OrderControllerTest extends \Wirecard\Test\WdUnitTestCase
+{
+    /**
+     * @var OrderController
+     */
+    private $_controller;
+
+    protected function setUp()
+    {
+        $this->_controller = oxNew(OrderController::class);
+
+        parent::setUp();
+    }
+
+    public function testInit()
+    {
+
+        $result = $this->_controller->init();
+        $this->assertTrue($result);
+    }
+
+    public function testInitFromRedirect()
+    {
+        oxTestModules::addFunction(
+            'oxUtils',
+            'redirect',
+            '{ return $aA[0]; }');
+
+        $_POST['wdpayment'] = "sessionToken";
+        Registry::getSession()->setVariable('wdtoken', "sessionToken");
+
+        $oUserStub = $this->getMockBuilder(User::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->_controller->setUser($oUserStub);
+
+        $result = $this->_controller->init();
+        $this->assertContains('wdtoken=sessionToken', $result);
+    }
+
+    public function testInitFromRedirectFromForm()
+    {
+        oxTestModules::addFunction(
```

Comment:
test testInitFromRedirectFromForm() and test testInitFromRedirect() are 98% same can we use a dataprovider to set the redirectFromForm or not.